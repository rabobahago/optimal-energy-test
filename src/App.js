import GridLayout from "./components/LayoutComponent";

function App() {
  return (
    <div className="App">
      <h1>Optimax Energy Challenge</h1>
      <GridLayout />
    </div>
  );
}

export default App;
