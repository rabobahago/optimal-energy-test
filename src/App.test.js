import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

it("should take a snapshot", () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});

describe("App component", () => {
  it("renders App component correctly", () => {
    const { getByText } = render(<App />);
    expect(getByText(/Optimax Energy Challenge/i)).toBeInTheDocument();
  });
});
