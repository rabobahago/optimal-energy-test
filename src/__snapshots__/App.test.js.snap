// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Optimax Energy Challenge
    </h1>
    <div>
      <button>
        Add Grid
      </button>
      <div
        class="react-grid-layout"
        style="height: 2910px;"
      >
        <div
          class="react-grid-item gridItem react-draggable cssTransforms react-resizable"
          data-grid="[object Object]"
          style="width: 453px; height: 1170px; position: absolute; transform: translate(20px,20px);"
        >
          <button
            class="removeButton"
          >
            x
          </button>
          <div>
            grid1
          </div>
          <span
            class="react-resizable-handle react-resizable-handle-se"
          />
        </div>
        <div
          class="react-grid-item gridItem react-draggable cssTransforms react-resizable"
          data-grid="[object Object]"
          style="width: 768px; height: 320px; position: absolute; transform: translate(493px,2230px);"
        >
          <button
            class="removeButton"
          >
            x
          </button>
          <div>
            grid2
          </div>
          <span
            class="react-resizable-handle react-resizable-handle-se"
          />
        </div>
        <div
          class="react-grid-item gridItem react-draggable cssTransforms react-resizable"
          data-grid="[object Object]"
          style="width: 610px; height: 320px; position: absolute; transform: translate(493px,2570px);"
        >
          <button
            class="removeButton"
          >
            x
          </button>
          <div>
            grid3
          </div>
          <span
            class="react-resizable-handle react-resizable-handle-se"
          />
        </div>
        <div
          class="react-grid-item gridItem react-draggable cssTransforms react-resizable"
          data-grid="[object Object]"
          style="width: 925px; height: 1000px; position: absolute; transform: translate(335px,1210px);"
        >
          <button
            class="removeButton"
          >
            x
          </button>
          <div>
            grid4
          </div>
          <span
            class="react-resizable-handle react-resizable-handle-se"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
